src/algorithms.rs

```diff
@@ -110,3 +111,44 @@ pub fn generate_multi_prime_key<R: Rng>(
         primes,
     ))
 }
+
+/// Mask generation function.
+///
+/// Panics if out is larger than 2**32. This is in accordance with RFC 8017 - PKCS #1 B.2.1
+pub fn mgf1_xor(out: &mut [u8], digest: &mut dyn DynDigest, seed: &[u8]) {
+    let mut counter = [0u8; 4];
+    let mut i = 0;
+
+    const MAX_LEN: u64 = u32::MAX as u64 + 1;
```

Comment:
This is causing the CI failure:
```suggestion
    const MAX_LEN: u64 = std::u32::MAX as u64 + 1;
```